embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/JdbcInputConnection.java

```diff
@@ -187,35 +190,69 @@ public String buildSelectQuery(String tableName,
-            } else {
-                String upperOrderByColumn = orderByColumn.get().toUpperCase();
-                String lowerOrderByColumn = orderByColumn.get().toLowerCase();
-                if (columnNames.contains(upperOrderByColumn)) {
-                    if (columnNames.contains(lowerOrderByColumn)) {
-                        throw new ConfigException(String.format("Cannot specify order-by colum '%s' because both '%s' and '%s' exist.",
-                                orderByColumn.get(), upperOrderByColumn, lowerOrderByColumn));
-                    } else {
-                        actualOrderByColumn = upperOrderByColumn;
-                    }
+        if (orderByColumn.isPresent() || incrementalColumn.isPresent()) {
+            String actualOrderByColumn = null;
+            if (orderByColumn.isPresent()) {
+                Set<String> columnNames = getColumnNames(actualTableName);
+                if (columnNames.contains(orderByColumn.get())) {
+                    actualOrderByColumn = orderByColumn.get();
                 } else {
-                    if (columnNames.contains(lowerOrderByColumn)) {
-                        actualOrderByColumn = lowerOrderByColumn;
+                    String upperOrderByColumn = orderByColumn.get().toUpperCase();
+                    String lowerOrderByColumn = orderByColumn.get().toLowerCase();
+                    if (columnNames.contains(upperOrderByColumn)) {
+                        if (columnNames.contains(lowerOrderByColumn)) {
+                            throw new ConfigException(String.format("Cannot specify order-by colum '%s' because both '%s' and '%s' exist.",
+                                    orderByColumn.get(), upperOrderByColumn, lowerOrderByColumn));
+                        } else {
+                            actualOrderByColumn = upperOrderByColumn;
+                        }
                     } else {
-                        actualOrderByColumn = orderByColumn.get();
+                        if (columnNames.contains(lowerOrderByColumn)) {
+                            actualOrderByColumn = lowerOrderByColumn;
+                        } else {
+                            actualOrderByColumn = orderByColumn.get();
+                        }
                     }
                 }
             }
 
-            sb.append("ORDER BY ").append(quoteIdentifierString(actualOrderByColumn)).append(" ASC");
+            sb.append(" ORDER BY ");
+            if (incrementalColumn.isPresent()) {
+                for (int i = 0; i < incrementalColumn.get().size(); i++) {
+                    String column = incrementalColumn.get().get(i);
+                    sb.append(quoteIdentifierString(column)).append(" ASC");
+                    if (i < incrementalColumn.get().size() - 1) {
+                        sb.append(", ");
+                    }
+                }
+            }
+            if (orderByColumn.isPresent() && actualOrderByColumn != null) {
```

Comment:
Sorting records using `order_by` here doesn't make sense because records in the results are not actually sorted by `order_by` option. Records are sorted by `incremental_columns`. `order_by` option is ignored from user's point of view.